data/mods/ssb/abilities.ts

```diff
@@ -100,6 +100,21 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
 		},
 	},
 
+	// Coolcodename
+	firewall: {
+		shortDesc: "Burns opponents that attempt to use status moves on this Pokemon; Status move immunity.",
+		name: "Firewall",
+		onTryHit(target, source, move) {
+			if (move.category === 'Status' && target !== source) {
+				if (!source.trySetStatus('brn', target)) {
+					this.add('-immune', target, '[from] ability: Good as Gold');
```

Comment:
```suggestion
				this.add('-immune', target, '[from] ability: Firewall');
				source.trySetStatus('brn', target));
```

Reply:
Usually if there's another effect the immunity message is not shown. Will fix me leaving good as gold on there though